File: baseproject-android/app/src/main/java/com/chang/template/activity/BaseMVPFragment.java
package com.chang.template.activity;

import android.os.Bundle;
import android.support.annotation.Nullable;

/**
 * Created by Howard Chang on 2016/11/29
 */
public abstract class BaseMVPFragment extends BaseFragment implements BaseView {

    private BasePresenter mPresenter;

    public abstract void onBaseActivityCreated();

    public void setPresenter(BasePresenter presenter) {
        mPresenter = presenter;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        if (mPresenter != null) {
            mPresenter.onActivityCreated();
            onBaseActivityCreated();
        }
    }

    @Override
    public void onDestroy() {

        if (mPresenter != null) {
            mPresenter.onDestroy();
        }

        super.onDestroy();
    }

}

File: baseproject-android/app/src/main/java/com/chang/template/activity/BaseMVPFragment.kt
package com.chang.template.activity

import android.os.Bundle
import android.util.Log

/**
 * Created by Howard Chang on 2016/11/29
 */
abstract class BaseMVPFragment : BaseFragment(), BaseView {

    lateinit var basePresenter: BasePresenter

    abstract fun onBaseActivityCreated()

    open fun setPresenter(presenter: BasePresenter) {
        basePresenter = presenter
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        basePresenter.onActivityCreated()
        onBaseActivityCreated()
    }

    override fun onDestroy() {
        basePresenter.onDestroy()
        super.onDestroy()
    }

}
